Unit tests are failing on darwin / OSX

**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
`make test` on darwin fails for two packages
```
--- FAIL: TestRestoreAllWaitOldIptablesRestore (0.00s)
	iptables_test.go:1096: expected success, got iptables unsupported on this platform
--- FAIL: TestRestoreAllGrabNewLock (0.00s)
	iptables_test.go:1149: expected to lock xtables.lock, got iptables unsupported on this platform
	iptables_test.go:1157: expected timeout error, got iptables unsupported on this platform
--- FAIL: TestRestoreAllGrabOldLock (0.00s)
	iptables_test.go:1187: expected to lock @xtables, got listen unix @xtables: bind: address already in use
FAIL
FAIL	k8s.io/kubernetes/pkg/util/iptables	0.025s
```
```
--- FAIL: TestLoopbackHostPort (0.00s)
	config_selfclient_test.go:41: expected host to start with 127., got "::1"
# ...
FAIL
FAIL	k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server	2.136s
```
**What you expected to happen**:
These tests should pass on darwin / OSX, or be skipped if they're linux-specific.  [Our docs on unit tests](https://github.com/kubernetes/community/blob/master/contributors/devel/testing.md#unit-tests) say they should pass on all platforms.

**How to reproduce it (as minimally and precisely as possible)**:
`make test` on darwin

**Anything else we need to know?**:
- The first failure looks like it just needs to be skipped on darwin, someone is welcome to take this on, I may not get to this right away
/sig network
- The second failure is addressed by https://github.com/kubernetes/kubernetes/pull/47203
/sig api-machinery

**Environment**:
- Kubernetes version (use `kubectl version`): master
- Cloud provider or hardware configuration**: macbook pro
- OS (e.g. from /etc/os-release): OS X 10.11.6
- Kernel (e.g. `uname -a`): `Darwin cloudtop.local 15.6.0 Darwin Kernel Version 15.6.0: Tue Apr 11 16:00:51 PDT 2017; root:xnu-3248.60.11.5.3~1/RELEASE_X86_64 x86_64`

--- a/pkg/util/iptables/iptables_test.go
+++ b/pkg/util/iptables/iptables_test.go
@@ -1,3 +1,5 @@
+// +build linux
+
 /*
 Copyright 2014 The Kubernetes Authors.
 

